### DISCUSSION TITLE
optuna get stucks at local minimum in some experiments

### DISCUSSION THREAD

**Discussion by astrung** [optuna/optuna#5092] [Q&A]
I have an LSTM model, and I use `optuna` to set up tuning for my model with `number of trials =100`. I have run it in several time to test whether I can get global minimum in search space. 
In most of running, I will got same result, so it means I have reached to global minimum in most cases. **But in some cases, I see my tuning progress stuck at local minimum (which is higher error that most of other experiments). 
So how can I avoid this situation if I only run one time experiment? If I need to run experiment to figure out which is local minimum and which is global minimum, it will cost so much time and cloud fee.**

Below images are some of my final logs in experiments. **As you can see, most of experiment will have minimum error = 0.044(global minimum), but in last line experiment, it stucks at minimum error = 0.057(local minimum). 
Can I have any solution to avoid last line case ? please help**
![all_log](https://github.com/optuna/optuna/assets/9162294/a49f7bc4-c074-4755-b545-7c5efb5cffd2)

I attached my code too. Sorry but my data is private, so I can't share it
[test.zip](https://github.com/optuna/optuna/files/13236325/test.zip)

**Comment by astrung** [comment]
sorry but can anyone help ???

**Comment by nzw0301** [comment]
In my poor experience on hyper-parameter search, there is not popular approach for this exploitation because the problem is non-convex.  Maybe increasing `n_startup_trials` mitigates this but it depends on your budget and objective.

https://github.com/optuna/optuna/issues/823#issuecomment-570028962 seems similar idea: perform random search for exploration to avoid local minima rather than exploitation.

**Comment by astrung** [comment]
@nzw0301 
thank you for your comment, but which parameters I need to change ? Or may be I need to use different libraries ?

**Reply by nzw0301** [reply]
Sampler's `n_startup_trials` argument, for example [`TPESampler`](https://optuna.readthedocs.io/en/stable/reference/samplers/generated/optuna.samplers.TPESampler.html).

**Reply by astrung** [reply]
@nzw0301 sorry but I tried to read [TPE document](https://optuna.readthedocs.io/en/stable/reference/samplers/generated/optuna.samplers.TPESampler.html), but I still can not get which parameters I need to config, may be `constant_liar` or `warn_independent_sampling` ?

**Reply by nzw0301** [reply]
> Sampler's n_startup_trials argument, for example [TPESampler](https://optuna.readthedocs.io/en/stable/reference/samplers/generated/optuna.samplers.TPESampler.html).
